linux/devdoc/file_linux_requirements.md

```diff
@@ -0,0 +1,159 @@
+# file_linux
+================
+
+## Overview
+
+Linux implementation of the `file` module.
+
+-`file_create` uses [`open`](https://www.man7.org/linux/man-pages/man2/open.2.html).
+-`file_destroy` uses [`close`](https://www.man7.org/linux/man-pages/man2/close.2.html).
+-`file_write_async` and `file_read_async` create [`sigevent`](https://man7.org/linux/man-pages/man7/sigevent.7.html)s to specify callbacks.
+-`file_write_async` uses [`aio_write`](https://man7.org/linux/man-pages/man3/aio_write.3.html).
+-`file_read_async` uses [`aio_read`](https://man7.org/linux/man-pages/man3/aio_read.3.html).

```

Comment:
what happened to "extend" API? t is not listed here.

Reply:
I will add the extend functionality on linux later.

---
In reply to: [450467409](https://github.com/Azure/azure-c-pal/pull/7#discussion_r450467409) [](ancestors = 450467409)